lib/surface/components/live_file_input.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Surface.Components.LiveFileInput do
+  @moduledoc """
+  Wrapper around Phoenix LiveView's built-in
+  `live_file_input/2` function.
+
+  The parent LiveView, or LiveComponent must `allow_uploads` during mount:
+
+  ```elixir
+    def mount(socket) do
+      socket = allow_upload(socket, :avatar, accept: ~w(.json), max_entries: 1)
+      {:ok, socket}
+    end
+  ```
+
+  See Phoenix.LiveView [Uploads documentation](https://hexdocs.pm/phoenix_live_view/uploads.html#content)
+  """
+
+  use Surface.Component
+  import Surface.Components.Form.Utils, only: [props_to_attr_opts: 2]
+
+  @doc "Upload specified via `allow_upload`"
+  prop upload, :struct, required: true
+
+  @doc "The id of the corresponding input field"
+  prop id, :string
```

Comment:
This will not work anymore with PhoenixLiveView `0.15.2`, see the following commit: https://github.com/phoenixframework/phoenix_live_view/commit/c263db32f446e57275f62f507f4995fef59c1975 we could simply remove it.

Reply:
good catch! 
I'll remove it.